sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_delete_polling.py

```diff
@@ -44,13 +43,13 @@ def run(self):
 
     def finished(self):
         # type: () -> bool
-        if self._sd_disabled:
+        if not self._deleted_resource.recovery_id:
```

Comment:
Seems to deserve a comment that `recovery_id` is falsey when the vault has soft-delete disabled. Also, how do you like `return not self._deleted_resource.recovery_id or self._status == "deleted"`?

Reply:
yeah i like the condensing, thanks!